packages/opa-react/src/use-authz.ts

```diff
@@ -25,76 +25,28 @@ export default function useAuthz(
   if (context === undefined) {
     throw Error("Authz/useAuthz can only be used inside an AuthzProvider");
   }
-  const { sdk, defaultPath, defaultInput, defaultFromResult } = context;
-
-  const [isLoading, setIsLoading] = useState(true);
-  const [result, setResult] = useState<Result>();
-  const [error, setError] = useState<Error>();
-
-  // TODO(sr): This feels wrong. Figure out just how wrong it is.
-  const requestMemo = useDeepCompareMemoize({
-    defaultPath,
-    defaultInput,
-    defaultFromResult,
-    input,
-    path,
-    fromResult,
-  });
-
-  const evaluate = useCallback<(signal: AbortSignal) => Promise<Result>>(
-    async (signal: AbortSignal): Promise<Result> => {
-      const {
-        defaultPath,
-        defaultInput,
-        defaultFromResult,
-        input,
-        path,
-        fromResult: fromR,
-      } = requestMemo;
-      const p = path ?? defaultPath;
-      const i = mergeInput(input, defaultInput);
-      const fromResult = fromR ?? defaultFromResult;
-      const fetchOptions = { signal };
-      return p
-        ? sdk.evaluate(p, i, { fetchOptions, fromResult })
-        : sdk.evaluateDefault(i, { fetchOptions, fromResult });
+  const { defaultPath, defaultInput, defaultFromResult, queryClient } = context;
+  const p = path ?? defaultPath;
+  const i = mergeInput(input, defaultInput);
+  const fromR = fromResult ?? defaultFromResult;
+
+  const {
+    // NOTE(sr): we're ignoring 'status'
+    data: result,
+    error,
+    isFetching: isLoading,
+  } = useQuery(
+    {
+      queryKey: [p, i],
+      meta: { fromResult: fromR },
```

Comment:
ℹ️ Putting `fromResult` winto the `queryKey` wasn't correct: as far as the "hash" (JSON.stringify) was concerned, all functions were `null`, and this to surprising test failures. Should we determine that it's not acceptable to have the `fromResult` function be exempt from the cache key, we can provide a better hash function that can capture functions.

Reply:
I haven't been following this package, what's the difference between fromResult vs. result (data: result)?